frame/ibc-transfer/src/lib.rs

```diff
@@ -0,0 +1,576 @@
+				amount,
+			});
+			Ok(())
+		}
+
+		#[frame_support::transactional]
+		#[pallet::weight(<T as Config>::WeightInfo::open_channel())]
+		pub fn open_channel(origin: OriginFor<T>, params: OpenChannelParams) -> DispatchResult {
+			ensure_root(origin)?;
+			let order: Order = (&params).try_into().map_err(|_| Error::<T>::InvalidParams)?;
+
+			let connection_id = connection_id_from_bytes(params.connection_id)
+				.map_err(|_| Error::<T>::InvalidParams)?;
+			let counterparty_port_id = port_id_from_bytes(params.counterparty_port_id)
+				.map_err(|_| Error::<T>::InvalidParams)?;
+			let counterparty = Counterparty::new(counterparty_port_id, None);
+			let channel_end = ChannelEnd::new(
+				State::Init,
+				order,
+				counterparty,
+				vec![connection_id],
+				Version::new(VERSION.to_string()),
+			);
+
+			let port_id = PortId::transfer();
+			let channel_id = T::IbcHandler::open_channel(port_id.clone(), channel_end)
+				.map_err(|_| Error::<T>::ChannelInitError)?;
+			Self::deposit_event(Event::<T>::ChannelOpened {
+				channel_id: channel_id.to_string().as_bytes().to_vec(),
+				port_id: port_id.as_bytes().to_vec(),
+			});
+			Ok(())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::set_pallet_params())]
+		pub fn set_pallet_params(origin: OriginFor<T>, params: PalletParams) -> DispatchResult {
+			ensure_root(origin)?;
+			<Params<T>>::put(params);
+			Self::deposit_event(Event::<T>::PalletParamsUpdated {
+				send_enabled: params.send_enabled,
+				receive_enabled: params.receive_enabled,
+			});
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn is_send_enabled() -> bool {
+			Params::<T>::get().send_enabled
+		}
+
+		pub fn is_receive_enabled() -> bool {
+			Params::<T>::get().receive_enabled
+		}
+
+		pub fn resgister_asset_id(
+			asset_id: <T as DeFiComposableConfig>::MayBeAssetId,
+			denom: Vec<u8>,
+		) {
+			IbcAssetIds::<T>::insert(asset_id, denom)
+		}
+	}
+}
+
+#[derive(Clone)]
+pub struct IbcCallbackHandler<T: Config>(PhantomData<T>);
+
+impl<T: Config> core::fmt::Debug for IbcCallbackHandler<T> {
+	fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+		write!(f, "ibc-transfer")
+	}
+}
+
+impl<T: Config> Default for IbcCallbackHandler<T> {
+	fn default() -> Self {
+		Self(PhantomData::default())
+	}
+}
+
+impl<T: Config + Send + Sync> Module for IbcCallbackHandler<T> {
+	fn on_chan_open_init(
+		&mut self,
+		_output: &mut ModuleOutputBuilder,
+		_order: Order,
+		_connection_hops: &[ConnectionId],
+		_port_id: &PortId,
+		_channel_id: &ChannelId,
+		_counterparty: &Counterparty,
+		_version: &Version,
+	) -> Result<(), Ics04Error> {
+		Ok(())
+	}
+
+	fn on_chan_open_try(
+		&mut self,
+		_output: &mut ModuleOutputBuilder,
+		_order: Order,
+		_connection_hops: &[ConnectionId],
+		_port_id: &PortId,
+		_channel_id: &ChannelId,
+		_counterparty: &Counterparty,
+		version: &Version,
+		_counterparty_version: &Version,
+	) -> Result<Version, Ics04Error> {
+		Ok(version.clone())
```

Comment:
Going to assume there can be revisions to ICS20's version number, lets investigate the version that ibc-rs actually implements and return it here.